--- ggrelay-1.0-rc5A/scripts/ggrelay.orig	2004-02-07 19:17:12.000000000 +0100
+++ ggrelay-1.0-rc5A/scripts/ggrelay	2004-02-07 20:23:18.000000000 +0100
@@ -3,28 +3,35 @@
 # chkconfig: - 90 25
 # description: Gadu-Gadu relaying agent (proxy) daemon
 
+[ -f /etc/ggrelay/ggrelay.conf ] || exit 0
+
 # Source function library.
 . /etc/rc.d/init.d/functions
 
+
 RETVAL=0
 
 start() {
-    echo -n $"Starting ggrelay: "
-    daemon --check ggrelay xargs ggrelay -d \
-		< /etc/ggrelay/ggrelay.conf
-    RETVAL=$?
-    echo
-    [ $RETVAL -eq 0 ] && touch /var/lock/subsys/ggrelay
+	if [ ! -f /var/lock/subsys/ggrelay ]; then
+	    msg_starting "GGrelay"
+	    daemon ggrelay -d /etc/ggrelay/ggrelay.conf
+	    RETVAL=$?
+	    [ $RETVAL -eq 0 ] && touch /var/lock/subsys/ggrelay
+	else
+		msg_already_running "GGrelay"
+	fi
 }
 
 stop() {
-    echo -n $"Stopping ggrelay: "
-    killproc ggrelay
-    RETVAL=$?
-    echo
-    if [ $RETVAL -eq 0 ]; then
-        rm -f /var/lock/subsys/ggrelay
-    fi
+	if [ -f /var/lock/subsys/ggrelay ]; then
+	  	msg_stopping "GGrelay"
+		busy
+		killproc ggrelay
+        	rm -f /var/lock/subsys/ggrelay > /dev/null 2>&1
+		ok
+   	else
+		msg_not_running "GGrelay"
+	fi	
 }
 
 # See how we were called.
